### DISCUSSION TITLE
unstable_FormInput as=textarea doesn't affect form state value

### DISCUSSION THREAD

**Discussion by awinogradov** [ariakit/ariakit#869] [Questions] [answered]
So, I have

``` ts
export const TextArea: React.FC<TextAreaProps> = (props) => (
    <Input as="textarea" style={{ minHeight: '100px' }} {...props} />
);
```

This is a wrapper for `unstable_FormInput`. It works well with `unstable_Form` and `unstable_useFormState`, but if I pass `as="textarea"` prop nothing happens with form state 🤷🏼‍♂️  Same with [docs](https://reakit.io/docs/form/#textareas) usage.

**Comment by diegohaz** [comment, marked as answer]
It's hard to debug with just that snippet of code. Could you put together a CodeSandbox reproducing the problem?

**Reply by awinogradov** [reply]
Yep, sure! https://codesandbox.io/s/silly-feather-wpcwg?file=/src/App.js
I found the problem. It happens only in the case of using SC for the `unstable_FormInput`.

**Reply by awinogradov** [reply]
The problem can be solved if wrap in this way

``` ts
const StyledInput = styled((props) => <FormInput as="textarea" {...props} />)``;
```

But this is not obvious 🤷🏼‍♂️

**Reply by diegohaz** [reply]
Ah, I see! That's because SC has its own `as` prop that's not passed to the underlying component. That is, when you do `<StyledInput as="textarea" />` you're essentially overriding the component passed to the `styled()` function.

You should do `<StyledInput forwardedAs="textarea" />` in this case so SC will pass `as="textarea"` to the Reakit `FormInput` component.

https://styled-components.com/docs/api#forwardedas-prop

**Reply by awinogradov** [reply]
Solved! Thanks @diegohaz ❤️
